packages/aws-rfdk/lib/core/lib/utils.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export class Utils {
+  /**
+   * This method compares two version strings
+   *
+   * @param versionFirst
+   * @param versionSecond
+   *
+   * @returns negative value if first version is smaller than second version;
+   * 0 if the versions matches, positive value if first version is greater
+   * than second version.
+   */
+  public static versionCompare(versionFirst: string, versionSecond: string): number {
+    const regExStripZero = /(\.0+)+$/;
+    const versionArrayFirst = versionFirst.replace(regExStripZero, '').split('.');
+    const versionArraySecond = versionSecond.replace(regExStripZero, '').split('.');
+    const minLen = Math.min(versionArrayFirst.length, versionArraySecond.length);
+
+    for (let i = 0; i < minLen; i++) {
+      const diff = parseInt(versionArrayFirst[i], 10) - parseInt(versionArraySecond[i], 10);
+      if (diff) {
+        return diff;
+      }
+    }
+    return versionArrayFirst.length - versionArraySecond.length;
```

Comment:
This function can currently be invoked as:

```ts
const result = versionCompare('10.1', '10.1.9');    // result will be -1
```

Is this the expected behavior? I suppose it depends on the intended semantic meaning when passing `"10.1"` as an argument to this function? Right now it is being treated as an implicit `"10.1.0"`, but I wonder if some users might consider `"10.1"` to mean the latest `10.1.x.y` release.

We cannot resolve versions (`10.1.x.y`) at deploy-time, so perhaps we should throw an exception if the two versions don't have the same number of components when being passed into this function. Callers can add `".0"` if need-be. What do you think?

Reply:
Modified the code to only accept full, resolved version string. We can add another method to parse partial, unresoled version string once we add that logic.